File: C++/Search_Algos.cpp
#include <iostream>
using namespace std;
int main(){
    //LINEAR SEARCH (for unsorted array)
    int A[10] = {2,4,6,8,3,7,1,3,5,0};
    int key1 = 3;
    int count = sizeof(A)/sizeof(A[0]);
    for (int i = 0; i < count; i++)
    {
        if (A[i] == key1)
        {
            cout << key1 << " at index " << i << "\n";
        }
        else if ( i == count - 1 && A[i] == key1){
            cout << "Element Not Present\n";
        }
    }

    //______________________________________________________
    //BINARY SEARCH (for sorted array)
    int B[10] = {1,4,6,8,9,10,25,45,67,89};
    count = sizeof(B)/sizeof(B[0]);
    int start = 0, end = count - 1;
    int mid, key2 = 10;
    while (start < end)
    {
        mid = start + (end - start)/2; //there maybe case where (start + end) exceed max number limit.
        if (B[mid] == key2){
            cout << key2 << " at index " << mid << "\n";
            break;
        }
        else if(B[mid] < key2){
            start = mid + 1;
        }
        else{
            end = mid - 1;
        }
    }
    
}

File: Python/Search_Algos.py
#Linear search (unsorted)
L = [1,5,7,3,4,7,2,6,7,1,5]
key = 7
for i in range(len(L)):
    if L[i] == key:
        print(key, "at index", i)
    elif i == len(L) - 1 and L[i] == key:
        print("Element not present")
print()

#_______________________________________________
#Binary Search (sorted)
L = [1,3,4,7,9,13,43,56,79,80]
key = 4
left = 0
right = len(L) - 1
while(left <= right):
    mid = left + (right - left)//2
    if L[mid] == key:
        print(key, "at index",mid)
        break
    elif L[mid] < key:
        left = mid+1
    elif L[mid] > key:
        right = mid - 1